demo-apps/cf-plugtest-server/src/main/java/org/eclipse/californium/plugtests/resources/DefaultTest.java

```diff
@@ -47,11 +48,14 @@ public void handleGET(CoapExchange exchange) {
 				request.getCode(), 
 				request.getMID()));
 
-		if (request.getToken().length > 0) {
+		if (request.getToken() != null) {
 			payload.append("\nToken: ");
-			StringBuilder tok = new StringBuilder(request.getToken()==null?"null":"");
-			if (request.getToken()!=null) for(byte b:request.getToken()) tok.append(String.format("%02x", b&0xff));
-			payload.append(tok);
+			if (request.hasEmptyToken()) {
+				payload.append("[]");
+			} else {
+				for (byte b : request.getTokenBytes())
+					payload.append(String.format("%02x", b & 0xff));
+			}
```

Comment:
I think we can replace by payload.append(request.getTokenString())

Reply:
OK